main/ejml-core/src/org/ejml/ops/ConvertMatrixType.java

```diff
@@ -36,12 +35,13 @@
     public static Matrix convert( Matrix matrix, MatrixType desired ) {
         Matrix m = null;
 
+        if (matrix.getType().equals(desired)) {
```

Comment:
I think the original behavior here was correct, but the JavaDoc needs to be updated to make explicit what the contract is. The logic is that in every case a new instance should be created. That way you know you're not modifying the original and you don't need to check for the special case where its the same matrix type..

Reply:
Agree, always making a copy is easier to handle for the user